### ISSUE TITLE
Duplicated findings in the output results

### ISSUE THREAD

**Issue by tilperion2** [semgrep/semgrep#3618]
**Describe the bug**
When running SemGrep scans I see duplication of results for some rules in my automated scans. 
Looks like this behaviour introduces around version 0.57.0

**To Reproduce**
See two examples:
https://semgrep.dev/s/Gwz1?version=0.60.0 - two similar/duplicated finding for the same line
https://semgrep.dev/s/Gwz1?version=0.56.0 - only one finding reported - correct one.

**Expected behavior**
Rule should not produce duplicated output targeting the same line. 

**Screenshots**
![Screenshot 2021-07-28 at 17 52 37](https://user-images.githubusercontent.com/16115976/127355063-0207d4c6-c250-42c3-beb4-2671c435a724.png)

**What is the priority of the bug to you?**

P0-P1 - we are blocked and unable to update to the latest SemGrep versions, but we are reverted to v0.56.0 and able to unblock our CI flow. 

**Environment**
My builds, deployments and semGrep dev: https://semgrep.dev/s/Gwz1

**Comment by IagoAbal** [comment]
(UPDATED) Might be a duplicate of #3614, but this can be reproduced in the Playground whereas #3614 supposedly does not.

**Comment by IagoAbal** [comment]
@tilperion2 do you have any other example of this issue that you can share?

**Comment by tilperion2** [comment]
Sure, here are some similar stuff:
https://semgrep.dev/s/P82v/?version=0.60.0 - duplication for line 20.
https://semgrep.dev/s/JjlP/?version=0.56.0 - no duplicated lines.
OR:
https://semgrep.dev/s/5J6j?version=0.60.0 - duplicated line 21

**Comment by IagoAbal** [comment]
OK it seems clear that your problems come from the equivalences we introduce for imported stuff. In the first example, `Cipher` can be referred as `Cipher` or as `javax.crypto.Cipher`, and we produce two matches, one for each.

**Comment by IagoAbal** [comment]
This sounds very similar to #3614 but here we can reproduce the issue in the Playground, whereas in #3614 the double findings supposedly do not occur in the Playground... so maybe these are not duplicates after all. Perhaps just related.

**Comment by IagoAbal** [comment]
@tilperion2 I consider this a bug and we should fix it, but I also must note that there are situations were Semgrep can appear to report "duplicates" that really aren't duplicates, e.g. here https://semgrep.dev/s/2b25 there is a different match per field in the record. Is it possible that you could handle this situation more gracefully on your side?

**Comment by tilperion2** [comment]
> @tilperion2 I consider this a bug and we should fix it, but I also must note that there are situations were Semgrep can appear to report "duplicates" that really aren't duplicates, e.g. here https://semgrep.dev/s/2b25 there is a different match per field in the record. Is it possible that you could handle this situation more gracefully on your side?

Thanks, Iago. 
In general It is ok for us to have different results for the same line, and we are handling this mostly correctly, I believe. 
But for our custom SemGrep rules we have test code and tests that should work as predicted. So for examples I`ve shown we are expecting that only particular lines should be in report, to assert our tests passed.
